src/main/java/cn/nukkit/block/BlockLeaves2.java

```diff
@@ -40,15 +41,21 @@ public int getId() {
                     toItem()
             };
         } else {
-            if ((int) ((Math.random()) * 200) == 0 && this.getDamage() == DARK_OAK) {
+            if (ThreadLocalRandom.current().nextInt(200) == 0 && (this.getDamage() & 0x03) == DARK_OAK) {
                 return new Item[]{
-                        new ItemApple()
+                        Item.get(Item.APPLE)
                 };
             }
-            if ((int) ((Math.random()) * 20) == 0) {
-                return new Item[]{
-                        new ItemBlock(new BlockSapling(), this.getDamage() & 0x03, 1)
-                };
+            if (ThreadLocalRandom.current().nextInt(20) == 0) {
```

Comment:
This looks like almost the same code in the other `BlockLeaves`. Could it be merged?